README.md

```diff
@@ -232,6 +232,61 @@ files = ["cache/*.db"]
 there will be no application mapped to the resultant bindle. Pass `-a bindle` to
 push to the Bindle server but not register it with Hippo.
 
+### Conditions
+
+Sometimes a module, or its associated files, may appear in different places depending on
+your build settings. For example, many languages support "debug" and "release" builds which
+put their output in different directories.
+
+To address this, you can make a HIPPOFACTS entry conditional, using the optional `condition`
+property:
+
+```toml
+[[handler]]
+route = "/"
+condition = "$build_mode != 'release'"
+name = "out/debug/birdbattle.wasm"
+
+[[handler]]
+route = "/"
+condition = "$build_mode == 'release'"
+name = "out/release/birdbattle.wasm"
+```
+
+A condition contains three parts:
+
+* The _condition variable_. Variable references begin with a `$` and may contain alphanumeric
+  characters and underscores.
+* The _comparison operator_. This must be `==` (true if the variable matches the value) or
+  `!=` (true if the variable _doesn't_ match the value).
+* The _match value_. Values must be enclosed in single quotes, and may contain alphanumeric
+  charaters, underscores, hyphens and periods.
+
+When you run `hippo prepare`, `hippo bindle` or `hippo push`, only entries whose `condition`
+evaluates to `true` are included in the generated bindle. By default, no variables are set,
+but you can set them using the `-c`/`--build-condition` option. You can provide this
+option multiple times. Each occurrence must have a value of the form `<name>=<value>`.
+If a variable isn't set, it doesn't match any value (even an empty value).
+
+For example, with the above fragment:
+
+* `hippo push .` would create a parcel from `out/debug/birdbattle.wasm` and map it to the `/` route.
+  (The `!=` condition is true because `build_mode` doesn't have a value.)
+* `hippo push . -c build_mode=debug` would also create a parcel from `out/debug/birdbattle.wasm` and
+  map it to the `/` route. (The `!=` condition is true because, although `build_mode` has a value,
+  that value doesn't match `'release'`.)
+* `hippo push . -c build_mode=release` would create a parcel from `out/release/birdbattle.wasm` (note the
```

Comment:
Hmm... I'm wondering if these flags should be set depending on what flags were provided to hippo-cli. For example, as a user if I'm running `hippo push -v production .`, should we assume that the user wants their `build_mode` set to `release` (or in this case, production)?

If I wanted to push a production bindle, now I'd have to run `hippo push . -v production -c build_mode=release` or `hippo push . -v production` depending on whether the HIPPOFACTS file needs that build_mode variable set or not. That seems a bit clunky...